src/server/utils/bucket_diff.js

```diff
@@ -454,5 +486,16 @@ class BucketDiff {
         return target_pos;
     }
 
+    /**
+     * @param {any[]} bucket_obj
+     */
+    _get_latest_delete_marker(bucket_obj) {
+        // Extract the latest delete marker from the bucket
+        const latest_delete_marker = bucket_obj.reduce(function(prev, current) {
+            return (new Date(prev.LastModified) > new Date(current.LastModified)) ? prev : current;
```

Comment:
Why do we need to check for the date? 
why not simply check the `IsLatest` field?